### PR TITLE
Display Member Contact Info Based on Accept/Reject Status

### PR DESCRIPTION
**Pull request by Monica-Coira**
## Description

- Removes a user from a group when they click the Reject button.
- Adds a user back into a group if they click the Accept button after having rejected the group previously to give the user flexibility.
- For each group, only displays the users that have accepted that group and the user that is currently logged in as "Members".
- Creates input fields for email address and phone number in the Signup Form and stores this data in the database.
- Displays each group member's profile picture, email address, and phone number in the Group Card.
- In the next PR, I will refactor the MatchByAvailability algorithm to reduce time complexity.

## Milestones
This works towards Milestone 8, which is that users can accept and reject the study groups they are matched into, and towards allowing users to contact their fellow study group members.

## Test Plan

https://github.com/user-attachments/assets/67b85585-d808-4803-bb61-f411f87695f7

Screenshot 2025-07-20 at 1 13 01 PM
Besides testing basic functionality, these are the corner cases I tested:

- Viewed a study group from multiple user accounts to ensure that only the users that have accepted a group and the user logged in are displayed as members of that group.
- Rejected a group from one account and viewed the group in another to make sure that the user that rejected the group is not visible as a member to other users.
- Accepted a group from one account after having rejected it in the past to ensure that the user is added back to the group as a member and is visible to all other users.
- Attempted to sign up without inputting a phone number or email address to ensure that an error message appears.
- Displayed the contact information for users without a profile picture to ensure that the default profile picture is displayed.
- Ensured that if there are no groups under a certain section, the heading for that section is no longer visible. For example, moved all groups out of the "Available" section to ensure that the "Available Groups" heading no longer appears on the page.

### CODE REVIEW

File: client/src/components/GroupCard.jsx [hunk 1]
```diff
@@ -39,15 +57,19 @@ const GroupCard = ({className, dayOfWeek, time, users, groupCompatibilityScore,
             <h5 className="group-card-users">Members:</h5>
             <div className="group-card-members">
                 {users.map((item, index) => {
-                    return (
-                        <div className="group-card-member" key={index}>{item}</div>
-                    )
+                    if (userInformation[item]){
+                        return (
+                            <MemberCard 
+                                key={index} name={userInformation[item][0].name} email={userInformation[item][0].email} phoneNumber={userInformation[item][0].phone_number} profilePicture={userInformation[item][0].profile_picture} fetchData={fetchData} id={userInformation[item][0].id}
```
[the remark this answers is not in the document]
Comment (Monica-Coira) [reply client/src/components/GroupCard.jsx:63]: Got it, just added the check on line 60 in PR #56.
